### PR TITLE
search-route: complete test coverage and spec verification

### LINKED ISSUE #1202: search-route: complete test coverage and spec verification
### Description

The existing tests in `route_search_handler_test.go` cover the basic happy path and a few validation cases. The following spec-required scenarios are missing:

| Missing test case | What to assert |
|---|---|
| **404 on no results** | HTTP 404, `code: 404`, `text: "resource not found"` (existing `TestRouteSearchHandlerNoResults` currently asserts 200) |
| **`limitExceeded` = true** | Set `maxCount=1` for a multi-match query, assert `data.limitExceeded == true` |
| **`limitExceeded` = false** | Query with large `maxCount`, assert `data.limitExceeded == false` |
| **`includeReferences=false`** | Assert `data.references.agencies` is empty |
| **Deterministic sort order** | Assert `data.list` is sorted by `shortName` |
| **Agency references sorted** | Assert `data.references.agencies` is sorted by `id` |
| **`maxCount=0`** | Assert HTTP 400 |
| **Non-numeric `maxCount`** (e.g. `"abc"`) | Assert HTTP 400 |
| **Missing `input` param entirely** | Assert HTTP 400 (no `input` key at all) |
| **Context cancellation** | Add `search-route` entry to `context_cancellation_test.go` |
| **Response envelope** | Assert `version`, `currentTime`, `code`, `text` are all present and correct |

This issue should be worked on **after** the behavior changes from Issues 1–5 are merged, since the tests validate that new behavior.

### PR DESCRIPTION
**Pull request by ARCoder181105**
## Summary

Completes the search-route endpoint review against the wiki spec. The endpoint's behavior was already correct — this fills in the missing test coverage and cleans up two small inconsistencies found along the way.

**Tests added**
- `maxCount` validation: `0`, non-numeric, and missing `input` entirely all return 400
- `limitExceeded` asserted in both directions
- Response envelope: `version`, `currentTime`, empty `references.routes`, `outOfRange`
- Context cancellation, using a past deadline so it asserts an exact 504 with no timing race

**Cleanup**
- `routeSearchHandler` now uses `nulls.StringOrEmpty` instead of hand-rolled `sql.NullString` unwrapping, matching every other route-building handler
- Folded the redundant `TestRouteSearchHandlerMaxCountBoundaries` into the new table-driven `maxCount` test
- Fixed a doc comment claiming lat/lon/radius filtering the handler never had

No behavior changes.

## Notes

The "404 on no results" case from #1202 is deliberately skipped. #1197 closed by updating the spec: returning 200 with an empty list is maglev's intended deviation from legacy, matching `search/stop`. Adding the 404 would be a regression.

Two things left for the reviewer to decide, both out of scope here:
- `routes_fts` indexes `id, agency_id, short_name, long_name, desc`, while the legacy Java index only covers `route_long_name` (falling back to the combined id when null) — so maglev matches a superset, e.g. routes hit only via `route_desc`.
- `searchStopsHandler` has the same incorrect geo-filtering doc comment; it belongs to the search-stop card.

Closes #1202.

## Test plan
- [x] `go vet -tags "sqlite_fts5 sqlite_math_functions" ./...`
- [x] `go vet -tags "purego" ./...`
- [x] `make test`
- [x] `go fmt ./...`

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved route-search results so missing optional route fields are returned consistently as empty values.
  * Enhanced validation for result-count limits, including boundary cases and invalid inputs.
  * Corrected limit-status reporting for truncated and complete result sets.

* **Tests**
  * Expanded coverage for response metadata, missing inputs, rate limiting, not-found responses, and request cancellation.

### CODE REVIEW

File: internal/restapi/context_cancellation_test.go [hunk 1]
```diff
@@ -76,15 +81,18 @@ func TestContextCancellationHandling(t *testing.T) {
...
 				statusCode == http.StatusUnauthorized || // API key validation happens first
 				statusCode == http.StatusBadRequest ||
 				statusCode == http.StatusInternalServerError ||
 				statusCode == http.StatusRequestTimeout ||
 				statusCode == http.StatusGatewayTimeout ||
-				statusCode == http.StatusNotFound,
-				"Expected status 200, 401, 404, 500, 408, or 504, got %d", statusCode)
+				statusCode == http.StatusNotFound ||
+				statusCode == http.StatusTooManyRequests,
+				"Expected status 200, 400, 401, 404, 429, 500, 408, or 504, got %d", statusCode)
```
[the remark this answers is not in the document]
Comment (ARCoder181105) [reply internal/restapi/context_cancellation_test.go]: @coderabbitai i think i addressed this

[the remark this answers is not in the document]
Comment (ARCoder181105) [reply internal/restapi/context_cancellation_test.go]: Checked this against the current code — the rate limiter isn't shared here, so the 429 path doesn't apply to this test.

`NewRestAPI` constructs a fresh limiter per instance (`internal/restapi/rest_api.go:15-20`), and `createTestApi(t)` calls it per test. `TestRouteSearchHandlerContextCancellation` gets its own limiter with `burst=5` and issues exactly one request, so it can't be rate limited.

The 429 was real in the earlier revision, where search-route was a 6th row in `TestContextCancellationHandling` — that test shares one API instance across its subtests, so the 6th request exceeded `burst=5`. The follow-up commit moved it into a dedicated test specifically to remove that shared budget.

Verified: `-count=50` passes with no flakes, there's no `t.Parallel()` in either file, and the handler is demonstrably reached (logs `request deadline exceeded` for `/api/where/search/route.json`) with an exact 504 assertion rather than a status grab-bag.

Keeping the mux-based form to match `TestBlockHandlerContextCancellation`, the existing pattern for handler cancellation tests — invoking the handler directly would also bypass the API-key and version middleware and lose route-wiring coverage.

**Review by Ahmedhossamdev — COMMENTED** [review]
LGTM!
